network_simulator/plot_tool.py

```diff
@@ -0,0 +1,300 @@
+import matplotlib.pyplot as plt
+import numpy as np
+
+from statistics import *
+
+class PlotTool(object):
+    """
+    Produces graph of different data structures.
+    """
+    def __init__(self):
+        pass
+
+    def gen_count_interval_list(self, lst):
+        """
+        Computes a tuple list of counts based on a timestamp list.
+
+        :param list lst: list of timestamps to be converted.
+        :return: list
+        """
+        curr_count = 0
+        curr_time = GRAPH_WINDOW_SIZE
```

Comment:
Refactor (shift+F6 in PyCharm on Linux) to "curr_window_end_time", since it seems like this is the endpoint of the windows you are using.